node_modules/col-assets/lib/api.js

```diff
@@ -824,18 +851,20 @@ var deleteAsset = module.exports.deleteAsset = function(ctx, id, callback) {
     return callback({'code': 400, 'msg': validationResult.error.details[0].message});
   }
 
-  // Only course instructors are allowed to delete an asset
-  if (!ctx.user.is_admin) {
-    log.error({'id': id}, 'Unauthorized to delete an asset');
-    return callback({'code': 401, 'msg': 'Unauthorized to delete an asset'});
-  }
-
   // Retrieve the asset that is being deleted
-  getAsset(ctx, id, null, function(err, asset) {
+  var opts = {
+    'includeWhiteboardUsages': true
+  };
+  getAsset(ctx, id, opts, function(err, asset) {
     if (err) {
       return callback(err);
     }
 
+    if (!canDeleteAsset(ctx.user, asset)) {
+      log.error({'id': id}, 'Unauthorized to delete an asset');
```

Comment:
Rather than just logging an `Unauthorized` message here, can you clarify that the asset can't be deleted because it has activity associated to it?